backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/view/SqlViewDataRepository.java

```diff
@@ -603,24 +604,35 @@ public <T> List<T> retrieveModelsByIds(
 			@NonNull final ViewId viewId,
 			@NonNull final DocumentIdsSelection rowIds,
 			@NonNull final Class<T> modelClass)
+	{
+		return retrieveModelsByIdsAsStream(viewId, rowIds, modelClass)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	@Override
+	@NonNull
+	public <T> Stream<T> retrieveModelsByIdsAsStream(
```

Comment:
@cp-ps  @dm-ps  small observation: IMHO I would prefer to call such method as `streamModelsByIds` or simply `streamModels` instead of `retrieve .. AsStream`.

(leave it like it is now)